src/main/scala/com/databricks/labs/mosaic/core/Mosaic.scala

```diff
@@ -37,7 +37,7 @@ object Mosaic {
             }
 
         val coreIndices = indexSystem.polyfill(carvedGeometry, resolution, Some(geometryAPI))
-        val borderIndices = indexSystem.polyfill(borderGeometry, resolution, Some(geometryAPI))
+        val borderIndices = indexSystem.polyfill(borderGeometry, resolution, Some(geometryAPI)).diff(coreIndices)
```

Comment:
Here I am doing a difference in set of cellIDs.
Those cellIDs are later converted in chips, and then the chips are combined with ++.
Combining two sets of chips with ++ won't work because it would compare the object references of the polygon and they will always be different.
So if we want to do the ++ with sets we would need to combine the two sets of cellIDs together and then run a single function to convert them to chips. That would trigger an intersection for all core chips, which would be expensive.

Reply:
I misread the logic to be done on chips rather than on cell IDs (longs). This is more than fine.